src/OpenTK.Mathematics/Vector/Vector3h.cs

```diff
@@ -105,45 +105,25 @@ public Vector3h(float x, float y, float z)
         /// <summary>
         /// Initializes a new instance of the <see cref="Vector3h"/> struct.
         /// </summary>
-        /// <param name="v">The <see cref="Vector3"/> to convert.</param>
-        public Vector3h(Vector3 v)
-        {
-            X = (Half)v.X;
-            Y = (Half)v.Y;
-            Z = (Half)v.Z;
-        }
-
-        /// <summary>
-        /// Initializes a new instance of the <see cref="Vector3h"/> struct.
-        /// </summary>
-        /// <param name="v">The <see cref="Vector3"/> to convert.</param>
-        public Vector3h(in Vector3 v)
-        {
-            X = (Half)v.X;
-            Y = (Half)v.Y;
-            Z = (Half)v.Z;
-        }
-
-        /// <summary>
-        /// Initializes a new instance of the <see cref="Vector3h"/> struct.
-        /// </summary>
-        /// <param name="v">The <see cref="Vector3d"/> to convert.</param>
-        public Vector3h(Vector3d v)
+        /// <param name="xy">The x and y components of the Vector3h.</param>
+        /// <param name="z">The z component of the Vector3h.</param>
+        public Vector3h(Vector2h xy, float z = default)
```

Comment:
I think we want another ctor that should take `(Vector2 xy, float z)`

Reply:
The reason why you want half vectors is for upload and data gains, in which case it makes the most sense to have ctors that take float values. 
But I can also agree that making a `Vector3` and just casting it to `Vector3h` is also a fine solution for this.